darts/utils/data/encoder_base.py

```diff
@@ -238,9 +239,46 @@ def _merge_covariate(
         """
         return covariate.stack(encoded) if covariate is not None else encoded
 
+    @staticmethod
+    def _drop_encoded_components(
+        covariate: Optional[TimeSeries], components: pd.Index
+    ) -> Optional[TimeSeries]:
+        """Avoid pitfalls: `encode_train()` or `encode_inference()` can be called multiple times or chanined.
+        Exclude any encoded components from `covariate` to generate and add the new encodings at a later time.
+        """
+        if covariate is None:
+            return covariate
+
+        duplicate_components = components[components.isin(covariate.components)]
+        # case 1: covariate only consists of encoded components
+        if len(duplicate_components) == len(covariate.components):
+            covariate = None
+        # case 2: covariate has also non-encoded components
+        elif not duplicate_components.empty:
```

Comment:
Shouldn't this be something like
```suggestion
        elif len(duplicate_components) < len(covariate.components):
```